test/wallaby/browser_test.exs

```diff
@@ -1,7 +1,100 @@
 defmodule Wallaby.BrowserTest do
-  use ExUnit.Case, async: true
+  use ExUnit.Case
+
+  import Wallaby.SettingsTestHelpers
 
   alias Wallaby.Browser
+  alias Wallaby.Session
+
+  defmodule EchoingDriver do
+    def start_link(test_pid) do
+      Agent.start_link(fn -> test_pid end, name: __MODULE__)
+    end
+
+    def visit(%Session{}, path) do
+      Agent.get(__MODULE__, fn (test_pid) ->
+        send test_pid, {:visit, path}
+      end)
+    end
+  end
+
+  describe "visit/2" do
+    setup do
+      {:ok, _} = EchoingDriver.start_link(self())
+      ensure_setting_is_reset(:wallaby, :base_url)
+    end
+
+    test "relative path without leading slash, base url with trailing slash" do
+      Application.put_env(:wallaby, :base_url, "http://example.com/")
+
+      session = %Session{driver: EchoingDriver}
+      Browser.visit(session, "form.html")
+
+      assert_received {:visit, "http://example.com/form.html"}
+    end
+
+    test "relative path with leading slash, base url no trailing slash" do
+      Application.put_env(:wallaby, :base_url, "http://example.com")
+
+      session = %Session{driver: EchoingDriver}
```

Comment:
couldn't we also create the session in the setup and pass it into the tests? Seems to be needed everywhere, no biggy though :)